code/components/extra-natives-five/src/PatchNetworkChecks.cpp

```diff
@@ -60,13 +63,50 @@ static HookFunction hookFunction([]()
 		hook::put<int32_t>(location, (intptr_t)g_dynamicDoorCreationDisabled - (intptr_t)location - 4);
 	}
 
+	// Replace netgame check in CScenarioConditionIsMultiplayerGame::"check"
+	// This patch was designed to make SP car jacking animations work in MP (CConditionalClipSet)
+	//
+	{
+		uint64_t* result = 0;
+		auto parCguBuildStructureFromStaticData = hook::pattern("48 8D 15 ? ? ? ? 45 33 C9 44 8B C0 48 8B CB E8");
+		for (size_t i = 0; i < parCguBuildStructureFromStaticData.size(); i++)
+		{
+			result = parCguBuildStructureFromStaticData.get(i).get<uint64_t>();
+			if (!result)
+				continue;
+
+			auto ptrStructureNameHash = hook::get_address<uint64_t>(result, 3, 7);
+			if (ptrStructureNameHash && *(uint32_t*)ptrStructureNameHash == 0x0DDA3A703/*CScenarioConditionIsMultiplayerGame*/)
+				break;
+		}
+
+		assert(result);
+
+		auto scCondCtor = hook::get_address<uint64_t>((uint64_t)result - 0x56, 3, 7);
+		assert(scCondCtor);
+
+		auto scCondMultiplayerVft = hook::get_address<uint64_t>(scCondCtor + 0x13, 3, 7);
+		assert(scCondMultiplayerVft);
+
+		//1604-2060 = 0x18; 2189-2699 = 0x20; 2802-* = 0x38
+		auto vftOffset = xbr::IsGameBuildOrGreater<2802>() ? 0x38 : (xbr::IsGameBuildOrGreater<2189>() ? 0x20 : 0x18);
+
+		auto scCondMultiplayerCheckFunction = *(uint64_t*)(scCondMultiplayerVft + vftOffset);
+		assert(scCondMultiplayerCheckFunction);
+		assert(*(uint32_t*)scCondMultiplayerCheckFunction == 0x0538c033/*xor eax, eax; part of the next instruction*/);
+
+		g_SingleplayerScenarioConditionsDisabled = (bool*)hook::AllocateStubMemory(1);
+		hook::put<int32_t>(scCondMultiplayerCheckFunction + 0x4, (intptr_t)g_SingleplayerScenarioConditionsDisabled - scCondMultiplayerCheckFunction - 0x8);
+	}
+
 	OnKillNetworkDone.Connect([]()
 	{
 		se::ScopedPrincipal principalScopeInternal(se::Principal{ "system.internal" });
 		enableFlyThroughWindscreen.GetHelper()->SetValue("false");
 		enablePlayerRagdollOnCollision.GetHelper()->SetValue("false");
 		enablePlayerJumpRagdollControl.GetHelper()->SetValue("false");
 		enableDynamicDoorCreation.GetHelper()->SetValue("false");
+		enableSingleplayerScenarioConditions.GetHelper()->SetValue("false");
```

Comment:
Unrelated to the review here, but I wonder if this could be simplified so that variables that should 'reset' per session don't have to be added in multiple different places.